invenio_app_rdm/redirector/resource.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio App RDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Invenio RDM redirector resource."""
+
+from flask import redirect, request
+from flask_resources import Resource, ResourceConfig, route
+from invenio_records_resources.services.base.config import ConfiguratorMixin, FromConfig
+
+
+class RedirectorResource(Resource):
+    """Redirector resource class."""
+
+    def __init__(self, config):
+        """Instantiates redirector resource, storing the redirection rules."""
+        super().__init__(config)
+        self.rules = self.config.rules
+
+    def create_url_rules(self):
+        """Generates a list of rules, based on flask_resources.route. View methods are generated using a factory."""
+        url_rules = []
+        for endpoint, rule in self.rules.items():
+            source = rule["source"]
+            url_rules.append(
+                route(
+                    method="GET",
+                    rule=source,
+                    view_meth=self.redirect_view_factory,
+                    endpoint=endpoint,
+                )
+            )
+        return url_rules
+
+    def redirect_view_factory(self):
+        """Factory to create views which redirect requests to a URL."""
+        bp_endpoint = request.url_rule.endpoint
+
+        # Strip BP name prefix from endpoint
+        endpoint_name = bp_endpoint.replace(f"{self.config.blueprint_name}.", "")
+        triggered_rule = self.rules[endpoint_name]
+        target = triggered_rule["target"]
+        redirect_url = target
+        code = 301
```

Comment:
By default it sets HTTP status code to 301 (permanently redirected). 

The redirection rule can overwrite it (see below).